### PR TITLE
Add a VS Code dev container

### LINKED ISSUE #69: Create a dev container for building ovsrpro

### PR DESCRIPTION
**Pull request by brobeson**
- I added the files for a VS Code dev container.
- I committed the incorrect *projects/README.md* file. See #79 for the bug this exposes.

closes #69

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  description="Ovsrpro Centos 7 Development Container" \
+  name="ovsrpro-dev"
+
+USER 0
+RUN set -eux; \
+  yum --assumeyes update; \
+  yum install --assumeyes \
+    epel-release \
+    # This repository gives us more modern versions of some software, such as Git.
+    https://packages.endpoint.com/rhel/7/os/x86_64/endpoint-repo-1.7-1.x86_64.rpm; \
```
Comment (bnhhughes) [inline .devcontainer/Dockerfile:13]: Thanks for this, by the way - I was still using an old version of git on my CentOS VM until you mentioned this one includes a newer version.

**Review by bnhhughes — APPROVED** [review]

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  yum install --assumeyes \
+    cmake3 \
+    git \
+    make \
+    wget; \
+  # yum --assumeyes clean all; \
+  ln -s cmake3 /usr/bin/cmake; \
```
Comment (swflb) [inline .devcontainer/Dockerfile]: Do you also want to add `ln -s ccmake3 /usr/bin/ccmake; `?

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  ln -s ctest3 /usr/bin/ctest; \
+  ln -s cpack3 /usr/bin/cpack; \
+  yum install --assumeyes \
```
Comment (swflb) [inline .devcontainer/Dockerfile]: Why do 2 separate install steps? I think you should combine this step with line 14.

File: .devcontainer/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    # Required for Qt5
+    bison \
+    expat-devel \
+    fontconfig-devel \
+    freetype-devel \
+    gperf \
+    libxcb-devel \
+    libxkbcommon-devel \
+    openssl-devel \
+    which \
```
Comment (swflb) [inline .devcontainer/Dockerfile:32]: I am surprised `which` is not installed by default on a RHEL or CentOS system.

File: .devcontainer/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    zlib-devel \
+    # Required by ZooKeeper
+    ant \
+    autoconf \
+    automake \
+    java-11-openjdk-devel \
+    libtool; \
```
Comment (swflb) [inline .devcontainer/Dockerfile:39]: do you not need these

- ncurses-devel
- ninja-build
- glibs
- openssl
- dbus
- libXScrnSaver 
- libXtst
- pciutils
- EGL
- javac
I thought those were all necessary. but perhaps some of those are necessary only for developing applications that depend on ovsrpro. ?

**Review by swflb — APPROVED** [review]

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (brobeson) [reply .devcontainer/Dockerfile]: Sure, I overlooked that one.

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (brobeson) [reply .devcontainer/Dockerfile]: I prefer to have related functionality close together.  Hence creating the symlinks soon after installing CMake. Also, the first yum install is just necessary tools; the second is dependencies for specific ovsrpro projects. I'll change it, though.

File: .devcontainer/Dockerfile (hunk 4 above)
Comment (brobeson) [reply .devcontainer/Dockerfile:32]: So was I the first time that I encountered that missing gem.

File: .devcontainer/Dockerfile (hunk 5 above)
Comment (brobeson) [reply .devcontainer/Dockerfile:39]: Nope. ncurses-devel, openssl, dbus, libXScrnSave, libXtst, pciutils, and EGL are not needed. Building ovsrpro does not work with Ninja, see #72.

glibs-devel might have solved the Qt build error, but I didn't know about that. I tried glibc-devel and glib-devel, neither of which worked. So I just removed `-glib` in #71

**Comment by swflb** [comment]
@brobeson thanks for the explanations

**Comment by swflb** [comment]
@brobeson I just remembered, if we are going to be using VSCode, we might want this added to the .gitignore file
```
*vscode
```

**Comment by brobeson** [comment]
> @brobeson I just remembered, if we are going to be using VSCode, we might want this added to the .gitignore file
> 
> ```
> *vscode
> ```

This is one that I think everyone should add to their global Git ignore file.

**Comment by swflb** [comment]
`vim` might be handy to install. ?

**Comment by brobeson** [comment]
> @brobeson thanks for the explanations

You're welcome. The changes will take awhile to push up. I need to rebuild the container, then rebuild ovsrpro to ensure it all still works.

**Comment by brobeson** [comment]
> `vim` might be handy to install. ?

The container's use case is with VS Code, so I don't see a need for an editor.

**Pull request merged**